scripts/setupPythonEnvironment.bash

```diff
@@ -109,10 +124,25 @@ then
         if [ -d "$p" ]
         then
             echo "  $p"
-            RES=$($PYTHON_TARGET -m pip install $p 2>&1)
+            RES=""
+            if $VERBOSE
+            then
+                RES=$($PYTHON_TARGET -m $PIP_CMD install $p)
+                echo $RES
+            else
+                RES=$($PYTHON_TARGET -m $PIP_CMD install $p 2>&1)
+            fi
+            
+            if [[ $RES =~ "HTTP error" ]]
+            then
+                echo "Could not access package dependencies:"
+                echo $RES
+                exit 1
+            fi
+
             if [[ $RES =~ "Error" ]]
```

Comment:
Is there an error code sent by `pip` when it fails?
If so, capturign the `$?` shell variable just after the call seems like the most reliable way to solve this.

Reply:
It seems like pip just uses the unspecified error code, so we will still need to do some string parsing to figure out the details of any failures:

```
  ERROR: HTTP error 404 while getting https://github.com/LLNL/ATS/archive/refs/tags/7.0.105_badurl.tar.gz
ERROR: Could not install requirement ats@ https://github.com/LLNL/ATS/archive/refs/tags/7.0.105_badurl.tar.gz from https://github.com/LLNL/ATS/archive/refs/tags/7.0.105_badurl.tar.gz (from geos-ats==0.2.0) because of HTTP error 404 Client Error: Not Found for url: https://codeload.github.com/LLNL/ATS/tar.gz/refs/tags/7.0.105_badurl for URL https://github.com/LLNL/ATS/archive/refs/tags/7.0.105_badurl.tar.gz
WARNING: You are using pip version 22.0.4; however, version 23.3.1 is available.
You should consider upgrading via the '/usr/tce/bin/python3 -m pip install --upgrade pip' command.
[sherman@quartz1148:geos_ats_package]$ echo $?
1
```

Reply:
(Because in some cases, we want to continue trying to install the package in a virtual environment if we detect an error)

Reply:
> (Because in some cases, we want to continue trying to install the package in a virtual environment if we detect an error)

What is the use case for that?
At first thought, I would kill the process at the first error popping (providing the complete error message and asking the end-user to fix the error).

Reply:
> > (Because in some cases, we want to continue trying to install the package in a virtual environment if we detect an error)
> 
> What is the use case for that? At first thought, I would kill the process at the first error popping (providing the complete error message and asking the end-user to fix the error).

That's reasonable.  This fall-back option (where we try again to build in a virtual environment) could be replaced with a request to the user to re-try with the correct arguments.

And TBH, with recent changes to pip defaults, the user-privilege errors that would require a virtualenv be created don't seem to show up often now.

Reply:
@TotoGaz - with regards to the HTTPS connectivity issues we've seen, do you think we should just throw an error and tell users to setup the tests on a machine with access to github?  Or, should we try to add an option to install the ATS package from a local source as a fall-back?  (This second option may need to happen in the setup script being modified here, since it isn't a common pattern for setup.cfg files.)

Reply:
Managing error and fallbacks, etc... is really complicated. Especially in shell 😨 
Providing a script that can do it for us is fine, but if it fails, I would print an URL on readthedocs (or a github issue) where we can detail workarounds and hints to help.
Now, I do not know what that implies in term of `cmake` dependencies. But maybe we can revisit some elements as well?

Reply:
@TotoGaz - I've updated the script and tried to simplify the processing/error checking.  I've also removed the (never used) code to create a miniconda environment, since I don't think that we should maintain this.